### PR TITLE
Target refactor

### LINKED ISSUE #355: Refactor Browser.js into seperate files
https://github.com/kblok/puppeteer/pull/2/commits/ffe5b63dbade012744f6a91a6f9a89b3615d5246

This is not much about the refactor itself but about the changes they made on many methods and the target constructor

### PR DESCRIPTION
**Pull request by kblok**
closes #355

### CODE REVIEW

File: lib/PuppeteerSharp/Target.cs [hunk 1]
```diff
@@ -41,7 +48,10 @@ internal Target(Browser browser, TargetInfo targetInfo)
         /// Gets the type. It will be <see cref="TargetInfo.Type"/> if it's "page" or "service_worker". Otherwise it will be "other"
         /// </summary>
         /// <value>The type.</value>
-        public string Type => _targetInfo.Type == "page" || _targetInfo.Type == "service_worker" || _targetInfo.Type == "browser" ? _targetInfo.Type : "other";
+        public string Type => _targetInfo.Type == "page" || _targetInfo.Type == "service_worker" || _targetInfo.Type == "browser" ?
```
Comment (Meir017) [inline lib/PuppeteerSharp/Target.cs]: is it too late to introduce an enum for this?

File: lib/PuppeteerSharp/Target.cs [hunk 2]
```diff
@@ -10,24 +12,29 @@ namespace PuppeteerSharp
     public class Target
     {
         #region Private members
-        private Browser _browser;
         private TargetInfo _targetInfo;
+        private string _targetId;
```
Comment (Meir017) [inline lib/PuppeteerSharp/Target.cs]: these should be readonly

File: lib/PuppeteerSharp/Target.cs (hunk 1 above)
Comment (kblok) [reply lib/PuppeteerSharp/Target.cs]: It's never too late :)

**Comment by kblok** [comment]
How about now @Meir017 ?

File: lib/PuppeteerSharp/Helpers/FlexibleStringEnumConverter.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+
+namespace PuppeteerSharp.Helpers
+{
+    internal class FlexibleStringEnumConverter : StringEnumConverter
```
Comment (Meir017) [inline lib/PuppeteerSharp/Helpers/FlexibleStringEnumConverter.cs]: I'm guessing this file exists for the `other` case

Comment (kblok) [reply lib/PuppeteerSharp/Helpers/FlexibleStringEnumConverter.cs]: Exactly. I tried with `StringEnumConverter` but it failed when it got the type  `background_page`, and I was unable to find the complete list of types.

**Pull request merged**